lib/matplotlib/ticker.py

```diff
@@ -2359,7 +2368,7 @@ def tick_values(self, vmin, vmax):
 
             if vmin <= 0.0 or not np.isfinite(vmin):
                 raise ValueError(
-                    "Data has no positive values, and therefore can not be "
+                    "Data has non-positive values, and therefore can not be "
```

Comment:
This overall logic does not seem 10% correct because we are only checking vmin here and just below we ensure that vmin and vmax are sorted.

We also only get this error if the locator is not attached to as axis as in that case we silently clip vmin.